File: 30daysOFcode_Php/day 20.php
<?php

$handle = fopen("php://stdin", "r");
fscanf($handle, "%d", $n);
$a_temp = fgets($handle);
$a = explode(' ', $a_temp);
$a = array_map('intval', $a);

function swap($a, $left, $right)
{
    $old = $a[$right];
    $a[$right] = $a[$left];
    $a[$left] = $old;

    return $a;
}

$numberOfSwaps = 0;
for ($i = 0; $i < $n; $i++) {
    for ($j = 0; $j < $n - 1 - $i; $j++) {
        if ($a[$j + 1] < $a[$j]) {
            $a = swap($a, $j, $j + 1);
            $numberOfSwaps++;
        }
    }
}

print('Array is sorted in ' . $numberOfSwaps . ' swaps.') . PHP_EOL;
print('First Element: ' . current($a)) . PHP_EOL;
print('Last Element: ' . end($a)) . PHP_EOL;

File: 30daysOFcode_Go/solution_set2/day 20.go
package main

import "fmt"

func main() {
	var a []int = []int{}
	var i, j, e, n int

	fmt.Scan(&n)

	for i = 0; i < n; i++ {
		fmt.Scan(&e)
		a = append(a, e)
	}

	totalSwaps := 0

	for i = 0; i < n; i++ {
		numberOfSwaps := 0

		for j = 0; j < n-1; j++ {
			if a[j] > a[j+1] {
				a[j], a[j+1] = a[j+1], a[j]
				numberOfSwaps++
				totalSwaps++
			}
		}

		if numberOfSwaps == 0 {
			break
		}
	}

	fmt.Printf("Array is sorted in %d swaps.\n", totalSwaps)
	fmt.Printf("First Element: %d\n", a[0])
	fmt.Printf("Last Element: %d\n", a[len(a)-1])
}
